nbodykit/algorithms/threeptcf.py

```diff
@@ -108,11 +111,23 @@ def callback(r, i, j, iprim=None):
         # compute multipoles for each primary
         for iprim in range(len(pos)):
             tree_prim = kdcount.KDTree(numpy.atleast_2d(pos[iprim]), boxsize=boxsize).root
-            tree_sec.enum(tree_prim, rmax, process=callback, iprim=iprim)
+            tree_sec.enum(tree_prim, rmax, process=callback, iprim=iprim, bunch=bunchsize)
 
             if self.comm.rank == largest_load and iprim % chunk_size == 0:
                 self.logger.info("%d%% done" % (10*iprim//chunk_size))
 
+        # combine alms into zeta;
+        # this cannot be done in the callback because
+        # it is a nonlinear function (outer product) of alm.
+        for (l, m) in alms:
+            alm = alms[(l, m)]
```

Comment:
I believe alms should be summed after each primary is treated, i.e. this piece of code should be inside the loop on primaries
for iprim in range(len(pos)): (l. 112), just after l. 114
(and alms should be initialized just before l. 114)